### ISSUE TITLE
TESTNET: batch transactions feature broken

### ISSUE THREAD

**Issue by jkrauska** [CodaProtocol/coda#3599]
Observed in seared-kobe testnet. 
Commit 3abcbc1 on branch master

Command
`coda advanced batch-send-payments -p ~/wallet-keys/testkey /tmp/batch-send-payments.sexp`

Where the sexp file looks like this:
```
(
  (
    (receiver tdNDyN9cvjoKtqdoxmPBtvHWRZSKAECkA42HtohtnJj9YWF7roGGXrya91uNq5Zdj9WD62u525VESWHbC3XdeZxRXHTMyzumJ14T1coCRA2h5bu8ECM4o8WtUvJ5wAboGURXmujxn6Dm2k)
    (amount 5)
    (fee 10)
  )
  (
    (receiver tdNDmKVFEpy2k8BpMRw9zaFL6L9Kij64hGSrDDwrQNrt7hMFGt6ZwzcA9KeHYhQsJLhAbmJyguUzNgBbtPx8TgepPnA9ZgqfzFDzKixTw3F58hh9SrYa85UQh4zz2xodCvngLJgMkn27PK)
    (amount 5)
    (fee 10)
  )
)
```
The first transaction in the sexp will be enqueued and go through.
The second does not.

I believe this is in part due gossip/trust changes?

Here's what another node observes:
```
{
  "timestamp": "2019-10-09 20:10:38.129593Z",
  "level": "Debug",
  "source": {
    "module": "Trust_system__Peer_trust",
    "location": "File \"lib/trust_system/peer_trust.ml\", line 118, characters 56-63"
  },
  "message": "Increasing trust for peer $peer due to action Trust_system.Actions.Sent_useful_gossip ($cmd). New trust is 0.000102.",
  "metadata": {
    "cmd": {
      "payload": {
        "common": {
          "fee": 10,
          "nonce": "2",
          "memo": "2pn2ah78GsSX48v2GiqShtqAXQQx1qi8aaMj4UU5jMcJATXDmpfnZc"
        },
        "body": [
          "Payment",
          {
            "receiver": "tdNDyN9cvjoKtqdoxmPBtvHWRZSKAECkA42HtohtnJj9YWF7roGGXrya91uNq5Zdj9WD62u525VESWHbC3XdeZxRXHTMyzumJ14T1coCRA2h5bu8ECM4o8WtUvJ5wAboGURXmujxn6Dm2k",
            "amount": 5
          }
        ]
      },
      "sender": "tdNEAV2Zak3182f2qHdxeknQCFC3qWweXqpBx2tHq3jX6CEb37CG8eayqhgSL4xDHXcPxURn9EHmitWE6m512buidc3c6ebdhSfJzcr45mogH9hqcGCL5D4ZXmEGyMY7h3PuLZyhfruCJg",
      "signature": "EXQTLM8Gg5j3ZnMLpGusDN5DCJgvUy6HrE6gY7QwJzRgNmNpW7mTggw8Gs2PmKynhooCxsnBQ85PMHqcHjrsFnAC1fCTMRiPf2ZSw5NWp892DCqH3RmjgDusUK8Uft9B3Bxfhdf1Qmgw3UUPk32kNyzUfzE787YDTKeWynxWVCkZAKJ52TjxZLxEwGTBYxYxqe5r5LmmCmivDXqNL5nCZyWzhndEzQLLgS8hvTzAbgDSLbq9WffjcK7WGfvMho8GRJ73gzTRyJDGm8uc"
    },
    "peer": "34.222.1.217",
    "pid": 14472
  }
}
```

**Comment by jkrauska** [comment]
On a fresh run to new keys never used before

```
(((receiver
    tdNECMscyi1jrn9Aydzx6qpzgnbk6Xcz5Gvd8jrB1cBqkGsbja9Qkvj2oRrHysctGbCjx6VWY6pHytvU96dwDvWx1a61KCtRbRQAMJM1QzCw6TcApBwGqfMmBSY98pCLMXkKEmg55nd5pD)
   (amount 123) (fee 10))
((receiver
    tdNDeAysdNdKZhmgtJYNrg4izJs3GaBA4i783kwdKrYC6FweaCL7pB21JxLtTPDGY3UpopWV1givGDq4k27VfCxmtNDkzpHiA4Bm125tyqwrTrX2Uj9z5F1W8TXzxXK36YmkvWpyMRkznZ)
   (amount 123) (fee 10))
```

first key logs hits
```$ grep tdNECMscyi1jrn9Aydzx6qpzgnbk6Xcz5Gvd8jrB1cBqkGsbja9Qkvj2oRrHysctGbCjx6VWY6pHytvU96dwDvWx1a61KCtRbRQAMJM1QzCw6TcApBwGqfMmBSY98pCLMXkKEmg55nd5pD ~/test-coda/coda.log  | jq .message
"Attempted to record trust action of ourselves: Trust_system.Actions.Sent_useful_gossip ($cmd)"
"Added transaction $user_command to transaction pool successfully"
"Broadcasting transaction pool diff over gossip net"
"Increasing trust for peer $peer due to action Trust_system.Actions.Sent_useful_gossip ($cmd). New trust is -0.048913."
"Broadcasting transaction pool diff over gossip net"
```

second key log hits
```
$ grep tdNDeAysdNdKZhmgtJYNrg4izJs3GaBA4i783kwdKrYC6FweaCL7pB21JxLtTPDGY3UpopWV1givGDq4k27VfCxmtNDkzpHiA4Bm125tyqwrTrX2Uj9z5F1W8TXzxXK36YmkvWpyMRkznZ ~/test-coda/coda.log | jq .message
"Attempted to record trust action of ourselves: Trust_system.Actions.Sent_useless_gossip (rejecting $cmd because of $reason)"
"Added transaction $user_command to transaction pool successfully"
```

so it gets lost at local txn pool before broadcast.

**Comment by jkrauska** [comment]
The second entry is being rejected for 'invalid nonce'.

The nonces for the first key and second key are sequential and coming right after eachother.
```
  "timestamp": "2019-10-09 20:45:41.247202Z",
          "nonce": "4",

  "timestamp": "2019-10-09 20:45:41.255009Z",
          "nonce": "5",
```

**Comment by enolan** [comment]
fixed in 43cb1d6
